src/Platforms/DB2/Types/BooleanType.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Doctrine\DBAL\Platforms\DB2\Types;
+
+use Doctrine\DBAL\Platforms\Types\BooleanType as PlatformBooleanType;
+
+class BooleanType extends PlatformBooleanType
```

Comment:
I don't believe there should be a class extension here. Instead, what I see from working on the ORM recently is that there are two types of types in the DBAL/ORM:
1. Platform level types (e.g. Number, String) that implement the logic of schema definition and introspection.
2. DBAL/ORM level types that define the mapping between PHP values and the ones stored in and fetched from the database.

Most of the use cases require implementing the type on only one of the two levels but for some of them (e.g. date-related types), require both.

We may want to try splitting the Type API into the two above and see if it improves anything.